File: MiniDOMBase/DMD/Sistema/Procedure/Handlers/BaseProceduraHandler.cs

namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Gestore base di una procedura automatizzata
        /// </summary>
        public abstract class BaseProceduraHandler 
            : IProceduraHandler
        {

            /// <summary>
            /// Inizializza i parametri standard della procedura
            /// </summary>
            /// <param name="procedura"></param>
            public abstract void InitializeParameters(CProcedura procedura);

            /// <summary>
            /// Esegue la procedura
            /// </summary>
            /// <param name="procedura"></param>
            public abstract void Run(CProcedura procedura);

            
        }
    }
}

File: MiniDOMBase/DMD/Sistema/Procedure/Handlers/BaseProceduraHandler.vb
Imports minidom

Partial Class Sistema


    Public MustInherit Class BaseProceduraHandler
        Implements IProceduraHandler

        Public MustOverride Sub InitializeParameters(procedura As CProcedura) Implements IProceduraHandler.InitializeParameters

        Public MustOverride Sub Run(procedura As CProcedura) Implements IProceduraHandler.Run

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class

End Class

